src/tests/src/solver/expressions/test_AstDOTStyleVisitor.cpp

```diff
@@ -0,0 +1,169 @@
+legend_EqualNode [ label =" EqualNode: 1"]
+legend_ComponentVariableNode [ label =" ComponentVariableNode: 1"]
+legend_SumNode [ label =" SumNode: 1"]
+legend_DivisionNode [ label =" DivisionNode: 1"]
+legend_VariableNode [ label =" VariableNode: 1"]
+legend_MultiplicationNode [ label =" MultiplicationNode: 1"]
+legend_NegationNode [ label =" NegationNode: 1"]
+legend_ParameterNode [ label =" ParameterNode: 1"]
+legend_PortFieldNode [ label =" PortFieldNode: 1"]
+legend_ComponentParameterNode -> legend_ComponentVariableNode [style=invis];
+legend_ComponentVariableNode -> legend_DivisionNode [style=invis];
+legend_DivisionNode -> legend_EqualNode [style=invis];
+legend_EqualNode -> legend_GreaterThanOrEqualNode [style=invis];
+legend_GreaterThanOrEqualNode -> legend_LessThanOrEqualNode [style=invis];
+legend_LessThanOrEqualNode -> legend_LiteralNode [style=invis];
+legend_LiteralNode -> legend_MultiplicationNode [style=invis];
+legend_MultiplicationNode -> legend_NegationNode [style=invis];
+legend_NegationNode -> legend_ParameterNode [style=invis];
+legend_ParameterNode -> legend_PortFieldNode [style=invis];
+legend_PortFieldNode -> legend_SubtractionNode [style=invis];
+legend_SubtractionNode -> legend_SumNode [style=invis];
+legend_SumNode -> legend_VariableNode [style=invis];
+}
+}
+)raw";
+
+    Node* literalNode = registry.create<LiteralNode>(-40.);
+    Node* negationNode = registry.create<NegationNode>(literalNode);
+    Node* parameterNode = registry.create<ParameterNode>("avogadro_constant");
+    Node* multiplicationNode = registry.create<MultiplicationNode>(negationNode, parameterNode);
+    Node* variableNode = registry.create<VariableNode>("atoms_count");
+    Node* divisionNode = registry.create<DivisionNode>(variableNode, multiplicationNode);
+    Node* portFieldNode = registry.create<PortFieldNode>("gasStation", "1149");
+    Node* sumNode = registry.create<SumNode>(divisionNode, portFieldNode);
+    Node* componentVariableNode = registry.create<ComponentVariableNode>("1150", "otherStation");
+    Node* componentParameterNode = registry.create<ComponentParameterNode>("1151", "otherConstant");
+    Node* subtractionNode = registry.create<SubtractionNode>(componentVariableNode,
+                                                             componentParameterNode);
+    Node* equalNode = registry.create<EqualNode>(subtractionNode, sumNode);
+    Node* literalNode2 = registry.create<LiteralNode>(53.);
+    Node* lessThanOrEqualNode = registry.create<LessThanOrEqualNode>(literalNode2, equalNode);
+    Node* literalNode3 = registry.create<LiteralNode>(54.);
+    Node* greaterThanOrEqualNode = registry.create<GreaterThanOrEqualNode>(literalNode3,
+                                                                           lessThanOrEqualNode);
+
+    return std::pair{expected_msg, greaterThanOrEqualNode};
+}
+
+BOOST_FIXTURE_TEST_CASE(tree_with_all_type_node, Registry<Node>)
+{
+    auto [expected_msg, expr] = buildAllTypeNode(*this);
```

Comment:
i don't see simplification, on the contrary here you define Fixture class with methods, data member to replace one Function

Reply:
If my reply addresses the issue, please mark this thread as resolved.

Reply:
This suggestion is about simplicity **for the test**, also and clarity : 
- the test looses a line and gets more simple
- **buildAllTypeNode(...)** function makes 2 responsibilities actually, and should be split in two. 
  That's the reason why :  
  - it has to return 2 things under the form of a **std::pair** containing different object.
  - the function's name does not describe accurately what it does : the name forgets the expected dot content definition
- **buildAllTypeNode(...)** takes the **this** as an argument. If you know it refers to **Registry** passed to test as a fixture, good for you. But it's not obvious to everyone, and one has to think to understand.

Reply:
no, no and no! buildAllTypeNode has not 2 responsabilities but only one : Create and expression and because it create that expressionthis is the only place where we can build the expected output!

Reply:
how one can guess a number chosen by another? coding is not magic

Reply:
we don't connect on this, but I still resolve this comment.